src/components/NavBar.tsx

```diff
@@ -2,12 +2,100 @@ import { Container, Nav, Navbar, Dropdown } from "react-bootstrap";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faBars } from "@fortawesome/free-solid-svg-icons";
 import { Link, useLocation } from "react-router-dom";
-import { RoutePaths } from "../common/Routes";
+import { RoutePaths, getFullPath } from "../common/Routes";
+import styled from "styled-components";
 
 interface NavBarProps {
   links: { text: string; href: string }[];
 }
 
+// styled-components styling for Navbar
+
+const CustomNavbar = styled(Navbar)`
+  background-color: #f5f5f5;
+  padding: 0.5rem 1rem;
+  animation: fadeInDown 1s ease-out;
+
+  @keyframes fadeInDown {
+    from {
+      opacity: 0;
+      transform: translate3d(0, -40px, 0);
+    }
+    to {
+      opacity: 1;
+      transform: translate3d(0, 0, 0);
+    }
+  }
+`;
+
+const BrandText = styled(Navbar.Brand)`
+  margin-right: auto;
+  font-size: x-large;
+  color: #222831;
+
+  &:hover {
+    color: #07889b;
+    font-weight: bold;
+  }
+`;
+
+const LogoImage = styled.img`
+  height: 45px;
+  margin-bottom: 5px;
+`;
+
+const DropdownIcon = styled.div`
+  color: #222831;
+  cursor: pointer;
+
+  &:hover {
+    color: #0375b4;
+    transform: scale(1.25);
+  }
+`;
+
+const CustomCollapse = styled(Navbar.Collapse)`
+  background-color: #f5f5f5;
+
+  @media (max-width: 992px) {
+    position: absolute;
+    top: 56px;
+    right: 0;
+    width: 100%;
+    max-width: 250px;
+  }
+
+  @media (max-width: 992px) {
+    display: flex;
+    flex-direction: column;
+    align-items: flex-end;
+    padding-right: 1rem;
+  }
```

Comment:
you have two `@media (max-width: 992px)`? Can they both be put up as one?

Reply:
Yes sorry my bad, i combined them into one!